app/main.py

```diff
@@ -21,8 +22,13 @@ def main():
             print("Invalid move, please try again")
             continue
 
-        # make the move and update the board
-        chess_board.make_move(move)
+        if chess_board.is_white_piece(move) is white_turn:
```

Comment:
Regarding the `chess_board.is_white_piece(move)` as a function, I'm thinking this is a "Piece" attribute.
So maybe we could introduce a new **Piece** model and add a static method:

```python
class Piece:
    @staticmethod
    def is_white(board: "ChessBoard", move: "PieceMove") -> bool:
        if board[move.from_pos].islower():
            return True
        elif board[move.from_pos].isupper():
            return False
```